File: app/src/main/java/com/uncmorfi/about/AboutDialog.java
package com.uncmorfi.about;

import android.app.Dialog;
import android.content.pm.PackageInfo;
import android.content.pm.PackageManager;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.v4.app.DialogFragment;
import android.support.v7.app.AlertDialog;
import android.view.View;
import android.widget.TextView;

import com.uncmorfi.R;

public class AboutDialog extends DialogFragment {

    public AboutDialog() {}

    @Override
    @NonNull
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        return createNewCardDialog();
    }

    public AlertDialog createNewCardDialog() {
        AlertDialog.Builder builder = new AlertDialog.Builder(getContext());

        View v = View.inflate(getContext(), R.layout.dialog_about, null);

        TextView versionText = (TextView) v.findViewById(R.id.version_name);
        versionText.setText(getVersionName());

        builder.setView(v);
        return builder.create();
    }

    private String getVersionName() {
        String versionName = "";
        try {
            PackageInfo packageInfo = getContext().getPackageManager()
                    .getPackageInfo(getContext().getPackageName(), 0);
            versionName = packageInfo.versionName;
        } catch (PackageManager.NameNotFoundException e) {
            e.printStackTrace();
        }
        return versionName;
    }
}

File: app/src/main/java/com/uncmorfi/about/AboutDialog.kt
package com.uncmorfi.about

import android.app.Dialog
import android.os.Bundle
import android.support.v4.app.DialogFragment
import android.support.v7.app.AlertDialog
import android.view.View
import com.uncmorfi.BuildConfig
import com.uncmorfi.R
import kotlinx.android.synthetic.main.dialog_about.view.*

class AboutDialog : DialogFragment() {

    private val version: String = BuildConfig.VERSION_NAME

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val builder = AlertDialog.Builder(requireActivity())

        val v = View.inflate(context, R.layout.dialog_about, null)
        v.versionName.text = version

        builder.setView(v)
        return builder.create()
    }
}
